lib/core/src/clientLogin.cpp

```diff
@@ -260,6 +303,20 @@ int clientLogin(
         } // if _scheme_override
     } // if client side auth
 
+    static constexpr auto irods_430 = irods::version{4, 3, 0};
+    const auto server_version = irods::to_version(_comm->svrVersion->relVersion);
+    if (!server_version) {
+        std::cerr << "Failed to get version from server ["
+                  << _comm->svrVersion->relVersion
+                  << "]\n";
```

Comment:
should these functions be writing to stdout/stderr at all?

Reply:
The authentication plugin should because this is where the password prompt is coming from for native authentication in the absence of an `.irodsA` file. Some of the other error messages (like this one) are only relevant should the calls be in a client (i.e. not a server).

I'll think about whether this should include a log message in addition to the error message. I think it's helpful for the user to know why the authentication failed.

Reply:
I'll instead add this to the `rError` stack in `_comm` and log it to capture the client and server sides of the conversation.

Reply:
sounds good to me.